File: LAB6/C++/lab6_2.cpp
#include <iostream>
#include <cmath>
using namespace std;

double myFunc() {
    int n;

    cout << "Enter number: ";
    cin >> n;

    double sum = 0;
    for (int i = 1; i <= n; i++) {
        double a = pow(-1, i + 1) / i;
        sum += a;
    }
    return sum;
}

int main() {

    double sum = myFunc();
    cout << "The result is  " << sum;

    return 0;
}

File: LAB6/PYTHON/lab6_2.py
def myFunc(n):

    global sum
    if n == 0:
        return
    else:
        tempSum = 0

        for k in range(1, n + 1):
            tempSum += ((-1) ** (k + 1)) / k
        sum += tempSum

        myFunc(n - 1)


total = 0
myFunc(5)

print(sum)